lib/integrations/typeform.ts

```diff
@@ -41,68 +45,88 @@ export class TypeformIntegration implements Integration {
 			handle: this.options.defaultUser,
 		});
 		const formResponse = (event.data.payload as any).form_response;
-		const formId = formResponse.form_id;
-		const responseId = formResponse.token;
-		const cardSlug = `user-feedback-${formId}-${responseId}`;
-		const formResponseMirrorId = `https://api.typeform.com/forms/${formId}/responses?included_response_ids=${responseId}`;
+
 		const username = /\s/.test(formResponse.hidden.user)
 			? null
 			: formResponse.hidden.user;
+
+		const email = /\s/.test(formResponse.hidden.email)
+			? formResponse.hidden.email
+			: (formResponse.answers.find((el) => el.type === 'email') || {}).email ||
+			  null;
+
+		const cardSlug =
+			`user-feedback-${formResponse.form_id}-${formResponse.token}`.toLowerCase();
+
 		const timestamp = new Date(formResponse.submitted_at).toISOString();
-		const questionsToProps = {
-			'How did you first hear about balenaCloud?':
-				'howDidYouFirstHearAboutBalenaCloud',
-			'How would you describe your role?': 'howWouldYouDescribeYourRole',
-			'Could you briefly describe your use case?':
-				'couldYouBrieflyDescribeYourUsecase',
-			'How has your experience been so far? What can we improve? We count on your honest feedback to make balenaCloud better.':
-				'howHasYourExperienceBeenSoFar',
-			'How likely are you to recommend balenaCloud to a friend or co-worker?':
-				'howLikelyAreYouToRecommendBalenaCloud',
-		};
-		const data = _.chain(
-			_.zip(formResponse.definition.fields, formResponse.answers),
-		)
-			.map((pair: any[]) => {
-				if (!Object.keys(questionsToProps).includes(pair[0].title)) {
-					// The only questions we currently support are the ones in questionsToProps keys.
-					// Any other question is ommited
-					return [];
-				}
-				for (const [title, props] of Object.entries(questionsToProps)) {
-					if (title === pair[0].title) {
-						return [props, pair[1][pair[1].type]];
-					}
-				}
-				return [];
-			})
-			.filter((item) => {
-				return _.size(item) > 0;
-			})
-			.fromPairs()
-			.assign({
-				mirrors: [formResponseMirrorId],
-				user: username,
-				status: 'open',
-				timestamp,
-			})
-			.value();
-		return [
+
+		const surveyData = await formResponseToContractData(
+			username,
+			timestamp,
+			formResponse,
+		);
+
+		let contactContract: Contract | null = null;
+
+		if (username) {
+			const contactId = await this.context.getContactId({
+				handle: username || '',
```

Comment:
The handle is not a suitable identifier for the user, we should use the email instead, otherwise there is a non-trivial chance that the response will be linked to the wrong contact